app/qml/map/MapWrapper.qml

```diff
@@ -586,57 +587,109 @@ Item {
   }
 
   MapFloatButton {
-    id: undoButton
+    id: backButton
 
-    // Find out if undo would collide with activeLayerButton button
-    // based on distance between them
-    function wouldCollideWithLayerBtn()
-    {
-      let undoBtnRightMostX = undoButton.x + undoButton.width
-      let layerBtnRightMostX = activeLayerButton.x
-      let distance = layerBtnRightMostX - undoBtnRightMostX
-      return distance < InputStyle.smallGap / 2
+    onClicked: {
+      if ( root.state === "edit" ) {
+        cancelEditDialog.open()
+      }
+      else if ( root.state === "record" ) {
+        cancelEditDialog.open()
+      }
     }
 
-     // let accBtnRightMostX = accuracyButton.x + accuracyButton.width
-     // let syncBtnLeftMostX = syncButton.x
-     // let distance = syncBtnLeftMostX - accBtnRightMostX
-     // return distance < InputStyle.smallGap / 2
-
-    onClicked: recordingToolsLoader.item.undo()
-
-    maxWidth: InputStyle.mapBtnHeight
-    withImplicitMargins: false
+    maxWidth: parent.width * 0.8
 
-    anchors.bottom: wouldCollideWithLayerBtn() ? activeLayerButton.top : parent.bottom
-    anchors.bottomMargin: root.mapExtentOffset + InputStyle.smallGap
+    anchors.top: howtoEditingBanner.showBanner ? howtoEditingBanner.bottom : parent.top
```

Comment:
We need to consider all banners, not just the `howToEditing` here:  
<img src="https://user-images.githubusercontent.com/22449698/191176051-d7247060-733f-44a1-91ed-b5eb88f285d4.png" width=500>